app/controllers/application_controller.rb

```diff
@@ -3,8 +3,161 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+  	enable :sessions 
+  	set :session_secret, "jbsalt"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do 
+  	erb :'index'
+  end 
+
+  get '/signup' do 
+  	if logged_in?
+  		redirect to '/tweets'
+  	else
+  		erb :'users/create_user' 
+  	end 
+  end 
+
+  post '/signup' do 
+  	# binding.pry
+  	if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.